test/cmocka/src/audio/volume/volume_process.c

```diff
@@ -250,7 +290,9 @@ static void verify_sX_to_s16(struct comp_dev *dev, struct comp_buffer *sink,
 				processed = INT16_MIN;
 
 			sample = (int16_t)processed;
-			assert_int_equal(dst[i + channel], sample);
+			delta = dst[i + channel] - sample;
+			if (delta > 1 || delta < -1)
+				assert_int_equal(dst[i + channel], sample);
```

Comment:
Shouldn't we have assert for `else` case? Idk if you can get delta in [-1:1], but even if we can you can add assert here, to log that something went really bad.

Reply:
The intent is to use assert_int_equal() to just fail the test. If the delta is 1, 0, or -1 the test for that sample is passed. I wasn't able to get the float code to round identically to FW code so I did this.